src/policy/feerate.h

```diff
@@ -38,10 +38,8 @@ class CFeeRate
 public:
     /** Fee rate of 0 satoshis per kvB */
     CFeeRate() : nSatoshisPerK(0) { }
-    template<typename I>
+    template<std::integral I>
```

Comment:
With all due respect, that's a semantically void comment. `std::integral I` already means "disallow floats."

Reply:
`std::integral` already means you specifically do not want to allow floats. If you did want to allow floats, you would have used `std::is_arithmetic_v`, but you didn't, so therefore you specifically want to exclude floats.

I'll concede the point that deleting the comment loses the information that you've previously had issues with silent conversions, although the presence of the constraint guarantees that you'll never have such issues again, so signaling about the previous issues would only serve the purpose of warning a future coder not to relax the constraint, which hopefully they would not do without thinking through the reason for its specificity.

Anyway, I will not be offended if anyone wants to edit this commit during merge. I only contributed it here because I was asked to send it upstream. All I cared about was that the Gentoo ebuilds for which I am responsible build okay with GCC 15. You all can do whatever you want with my suggested fixes.

Reply:
> You all can do whatever you want with my suggested fixes.

Thanks. We don't edit commits during merges, so I'll cherry-pick and fixup any nits in a new PR.